docs/source/common/optimizers.rst

```diff
@@ -516,3 +516,49 @@ to perform a step, Lightning won't be able to support accelerators and precision
     ):
         optimizer = optimizer.optimizer
         optimizer.step(closure=optimizer_closure)
+
+-----
+
+Configure Gradient Clipping
+---------------------------
+To configure custom gradient clipping, consider overriding
+the :meth:`~pytorch_lightning.core.lightning.LightningModule.configure_gradient_clipping` function.
+By default it will get arguments :attr:`~pytorch_lightning.trainer.Trainer.gradient_clip_val` and
+:attr:`~pytorch_lightning.trainer.Trainer.gradient_clip_algorithm` from ``Trainer`` in their respective
+fields and lightning will handle gradient clipping on its own. In case you want to set different values
+for your arguments of your choice and let Lightning handle the gradient clipping, you can use the inbuilt
+:meth:`~pytorch_lightning.core.lightning.LightningModule.clip_gradients` function and pass the arguments
+along with your optimizer.
+
+.. note::
+    Make sure to not override :meth:`~pytorch_lightning.core.lightning.LightningModule.clip_gradients`
+    function. If you want to customize gradient clipping, consider using
+    :meth:`~pytorch_lightning.core.lightning.LightningModule.configure_gradient_clipping` function.
+
+For example, here we will apply gradient clipping only to optimizer A.
+
+.. testcode:: python
+
+    def configure_gradient_clipping(self, optimizer, optimizer_idx, gradient_clip_val, gradient_clip_algorithm):
+        if optimizer_idx == 0:
+            # Lightning will handle the gradient clipping
+            self.clip_gradients(
+                optimizer, gradient_clip_val=gradient_clip_val, gradient_clip_algorithm=gradient_clip_algorithm
+            )
+
+Here we configure gradient clipping differently for optimizer B.
+
+.. testcode:: python
+
+    def configure_gradient_clipping(self, optimizer, optimizer_idx, gradient_clip_val, gradient_clip_algorithm):
+        if optimizer_idx == 0:
+            # Lightning will handle the gradient clipping
+            self.clip_gradients(
+                optimizer, gradient_clip_val=gradient_clip_val, gradient_clip_algorithm=gradient_clip_algorithm
+            )
+        elif optimizer_idx == 1:
+            custom_gradient_clip_val = 1e-2
+            for pg in optimizer.param_groups:
+                for p in pg["params"]:
+                    p.grad[p.grad > custom_gradient_clip_val] = custom_gradient_clip_val
+                    p.grad[p.grad <= 0] = 0
```

Comment:
that is a very funny way of clipping gradients haha
it will actually change the direction of the gradient
SGD will not be happy about this lol

Reply:
In all seriousness, I wouldn't put such an example.
maybe something like this could be more useful:

N = 2
# less clipping for parameters of optimizer 1
elif optimizer_idx == 1:
       self.clip_gradients(
                optimizer, gradient_clip_val=(N * gradient_clip_val), gradient_clip_algorithm=gradient_clip_algorithm
            )

or choose a different algorithm